### ISSUE TITLE
Pass arguments to scripts run via `sanity exec`

### ISSUE THREAD

**Issue by baybars1223** [sanity-io/sanity#2765]
**Is your feature request related to a problem? Please describe.**
I intend to have my migration script write a summary to file when it is finished.
I had hoped to pass the desired filename in by appending it to the `sanity exec` command, similar to how Yarn handles it. As far as I can tell, Sanity tosses any additional arguments.  
Looking at the `exec` code, I can't tell if this is intentional. In case it is, I labeled this a feature request.

**Describe the solution you'd like**
It would be nice if `sanity exec` could forward command line args to the script it executes.

**Describe alternatives you've considered**
I could, and likely will, use an environment variable, config file, or console.log the output instead and redirect it to a file with `>`. So, at least for my use case, this probably isn't a high-priority feature. Still, it would be good to have it at some point.

**Additional context**
Example command: `yarn sanity exec --with-user-token migration_scripts\test-sanity-exec-args.js foo.txt`
I added some logging to `@sanity\core\src\actions\exec\execScript.js` when I was trying to figure out if it was already possible.
Function `execScripts`'s first parameter, `args`, came through as:
```
{
    "groupOrCommand": "exec",
    "coreOptions": {
      "h": false,
      "help": false,
      "d": false,
      "debug": false,
      "v": false,
      "version": false
    },
    "argv": [
      "C:\\Program Files\\nodejs\\node.exe",
      "C:\\Users\\user\\dev\\repo\\node_modules\\@sanity\\cli\\bin\\sanity",
      "exec",
      "--with-user-token",
      "migration_scripts\\test-sanity-exec-args.js",
      "foo.txt"
    ],
    "extOptions": {
      "with-user-token": "migration_scripts\\test-sanity-exec-args.js"
    },
    "argsWithoutOptions": [
      "foo.txt"
    ],
    "extraArguments": []
  }
```

Function `parseCliFlags` returned:
```
{
  "_": [
    "exec",
    "foo.txt"
  ],
  "with-user-token": true,
  "mock-browser-env": false,
  "$0": "node_modules\\@sanity\\cli\\bin\\sanity"
}
```

**Comment by rexxars** [comment]
We use the convention of separating the command flags from script flags by a `--`, so for instance:

```
sanity exec --with-user-token myscript.js -- --some-script-flag
```

We _could_ pass on either unknown flags or all flags, I'm just a little worried about collisions and differing flag types.

**Comment by baybars1223** [comment]
Reasonable concerns.
I remember coming across the `--` somewhere, but it's possible that I only tried something like `sanity exec --with-user-token myscript.js -- some-script-flag`.
I'll double-check, thanks!

Regardless, is it possible we can get a note on the ` -- ` operator added to https://www.sanity.io/docs/exec and `sanity exec --help`? Now that you've mentioned it I can see it's used by a number of utilities and languages, but I don't think I'll be the only dev that hasn't come across it previously.

**Comment by ninaandal** [comment]
Thank you for reporting, @baybars1223! An explanation is included in `v2.30.0`.
